### ISSUE TITLE
No module named 'improved_diffusion'

### ISSUE THREAD

**Issue by liuyisu** [openai/improved-diffusion#123]
请问遇到ModuleNotFoundError: No module named 'improved_diffusion'
​该怎样处理呢

**Comment by tobi-ore** [comment]
clone the repo and install 'improved_diffusion' using: 
pip install -e .

**Comment by liuyisu** [comment]
> clone the repo and install 'improved_diffusion' using: pip install -e .
<img width="749" alt="58b478226e30bd7be28b95b4ecc65f8" src="https://github.com/openai/improved-diffusion/assets/155950168/1825df0c-4da3-4368-9b9d-a6ecd6ad3a0d">
Thank you very much for your reply. I encountered the above problem while testing on the server. Do you know how to handle it? Thank you very much

**Comment by tobi-ore** [comment]
Refer to the documentation provided in the repo on how to properly run the code. For example the image_train.py is in the scripts folder. You ought to use scripts/image_train.py

**Comment by liuyisu** [comment]
Yes, I later discovered and corrected this issue, it seems to be still improved_ The issue of diffusion. I am a beginner, this is really too difficult
<img width="567" alt="e1dde001131a1764e6432ee40ef333b" src="https://github.com/openai/improved-diffusion/assets/155950168/8891643c-6091-424b-97a1-499d62794eb4">

**Comment by tobi-ore** [comment]
Your `pip install -e` call is wrong. There is a full stop “.” after the -e to install the improved_diffusion package from the current directory `pip install -e .`. Make sure you’re in the improved_diffusion directory you cloned from here. Or put the path to the improved_diffusion folder instead of the “.”

**Comment by ali-alhwayzee** [comment]
There are two possibilities, according to what happened to me:

The first is to ensure that MPI is installed
[https://www.microsoft.com/en-us/download/details.aspx?id=105289](https://github.com/openai/improved-diffusion/issues/url)
The second is that the path is inside the "improved_diffusion" file
After setting the parameters, use the following command inside the terminal in the path above
python scripts/image_train.py --data_dir path/to/images $MODEL_FLAGS $DIFFUSION_FLAGS $TRAIN_FLAGS 
Make sure to check the image file path
"--data_dir path/to/images"

**Comment by gao040608** [comment]
你需要在当前环境下运行pip install -e .
